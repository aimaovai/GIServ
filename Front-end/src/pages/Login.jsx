import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import logo2 from "../images/logo2.png";

export default function Login() {
  // const navigate = useNavigate();

  const [details, setDetails] = useState({
    id: null,
    passwordd: null,
  });

  const login = () => {
    if (details.id === "1" && details.password === "admin") {
      localStorage.setItem("loggedin", "true");
      location.href = "/dashboard";
    } else {
      alert("wrong password");
    }
  };

  return (
    <div data-testid="login" className="h-screen bg-gradient-to-b from-red-500 to-purple-700 w-full flex justify-center items-center">
      <div className="flex flex-col gap-6 bg-neutral-900 p-20 text-center rounded-lg">
        <div className="flex h-16 justify-center">
          <img src={logo2} />
        </div>
        <p className="text-white">Dashboard Login</p>
        <input
          placeholder="Id"
          onChange={(e) => setDetails({ ...details, id: e.target.value })}
          className="h-10 outline-none bg-neutral-700 text-center text-white rounded-md"
        />
        <input
          type="password"
          placeholder="password"
          onChange={(e) => setDetails({ ...details, password: e.target.value })}
          className="h-10 outline-none bg-neutral-700 text-center text-white rounded-md"
        />
        <button
          className="bg-indigo-900 h-10 rounded-md text-white text-sm text-center"
          onClick={() => login()}
        >
          Login
        </button>
      </div>
    </div>
  );
}
