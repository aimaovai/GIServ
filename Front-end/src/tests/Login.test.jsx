/* 
  - Test suite id: TS_02
  - Test written by: Aima Ovai
  - Test written on: April 7, 2022
  - Test last updated on: Aprill 28, 2022
  - Test Description: This test suite is written to test the fucntionality
    of the login page. It tests that the page renders properly, and the
    login button acts as expected on click.
*/


import React from "react";
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, configure } from "enzyme";
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';

import Login from '../pages/Login';


configure({adapter: new Adapter()});

describe('GIServ Login Component', () => {
  
  // Test case id: TC_02
  it("Login page renders without crashing", async () => {
    try {
      //render loging page
      render(<Login />)
      const loginItem = await waitFor(() => screen.getByTestId('login'));
      expect(loginItem).toBeInTheDocument();
    } catch (e) {
      console.error('Failed to render login page without crashing.')
    }
  });

  // Test case id: TC_03
  test("Login page renders correctly", async () => {
    try {
      //render the login page
      render(<Login />)

      const loginItem = await waitFor(() => screen.getByTestId('login'));
      expect(loginItem).toBeInTheDocument();
    } catch (e) {
      console.error('Failed to render login page correctly.')
    }
  });

  // Test case id: TC_04
  test('Tests the login button on click', () => {
    try {
      const mockCallBack = jest.fn();
      const page = shallow(<Login />);
      page.find('button').simulate('click');
      expect(mockCallBack.mock.calls.length).not.toBeNull();
    } catch(e) {
      console.error('Failed to call action on button click.')
    }
  });
});
