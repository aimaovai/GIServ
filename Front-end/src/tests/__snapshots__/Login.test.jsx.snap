// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIServ Login Component Login page renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-screen bg-gradient-to-b from-red-500 to-purple-700 w-full flex justify-center items-center"
        data-testid="login"
      >
        <div
          class="flex flex-col gap-6 bg-neutral-900 p-20 text-center rounded-lg"
        >
          <div
            class="flex h-16 justify-center"
          >
            <img
              src="[object Object]"
            />
          </div>
          <p
            class="text-white"
          >
            Dashboard Login
          </p>
          <input
            class="h-10 outline-none bg-neutral-700 text-center text-white rounded-md"
            placeholder="Id"
          />
          <input
            class="h-10 outline-none bg-neutral-700 text-center text-white rounded-md"
            placeholder="password"
          />
          <button
            class="bg-indigo-900 h-10 rounded-md text-white text-sm text-center"
          >
            Login
          </button>
        </div>
      </div>
    </div>
    <div>
      <div
        class="h-screen bg-gradient-to-b from-red-500 to-purple-700 w-full flex justify-center items-center"
        data-testid="login"
      >
        <div
          class="flex flex-col gap-6 bg-neutral-900 p-20 text-center rounded-lg"
        >
          <div
            class="flex h-16 justify-center"
          >
            <img
              src="[object Object]"
            />
          </div>
          <p
            class="text-white"
          >
            Dashboard Login
          </p>
          <input
            class="h-10 outline-none bg-neutral-700 text-center text-white rounded-md"
            placeholder="Id"
          />
          <input
            class="h-10 outline-none bg-neutral-700 text-center text-white rounded-md"
            placeholder="password"
          />
          <button
            class="bg-indigo-900 h-10 rounded-md text-white text-sm text-center"
          >
            Login
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-screen bg-gradient-to-b from-red-500 to-purple-700 w-full flex justify-center items-center"
      data-testid="login"
    >
      <div
        class="flex flex-col gap-6 bg-neutral-900 p-20 text-center rounded-lg"
      >
        <div
          class="flex h-16 justify-center"
        >
          <img
            src="[object Object]"
          />
        </div>
        <p
          class="text-white"
        >
          Dashboard Login
        </p>
        <input
          class="h-10 outline-none bg-neutral-700 text-center text-white rounded-md"
          placeholder="Id"
        />
        <input
          class="h-10 outline-none bg-neutral-700 text-center text-white rounded-md"
          placeholder="password"
        />
        <button
          class="bg-indigo-900 h-10 rounded-md text-white text-sm text-center"
        >
          Login
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`GIServ Login Component Login renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-screen bg-gradient-to-b from-red-500 to-purple-700 w-full flex justify-center items-center"
      >
        <div
          class="flex flex-col gap-6 bg-neutral-900 p-20 text-center rounded-lg"
        >
          <div
            class="flex h-16 justify-center"
          >
            <img
              src="[object Object]"
            />
          </div>
          <p
            class="text-white"
          >
            Dashboard Login
          </p>
          <input
            class="h-10 outline-none bg-neutral-700 text-center text-white rounded-md"
            placeholder="Id"
          />
          <input
            class="h-10 outline-none bg-neutral-700 text-center text-white rounded-md"
            placeholder="password"
          />
          <button
            class="bg-indigo-900 h-10 rounded-md text-white text-sm text-center"
          >
            Login
          </button>
        </div>
      </div>
    </div>
    <div>
      <div
        class="h-screen bg-gradient-to-b from-red-500 to-purple-700 w-full flex justify-center items-center"
      >
        <div
          class="flex flex-col gap-6 bg-neutral-900 p-20 text-center rounded-lg"
        >
          <div
            class="flex h-16 justify-center"
          >
            <img
              src="[object Object]"
            />
          </div>
          <p
            class="text-white"
          >
            Dashboard Login
          </p>
          <input
            class="h-10 outline-none bg-neutral-700 text-center text-white rounded-md"
            placeholder="Id"
          />
          <input
            class="h-10 outline-none bg-neutral-700 text-center text-white rounded-md"
            placeholder="password"
          />
          <button
            class="bg-indigo-900 h-10 rounded-md text-white text-sm text-center"
          >
            Login
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-screen bg-gradient-to-b from-red-500 to-purple-700 w-full flex justify-center items-center"
    >
      <div
        class="flex flex-col gap-6 bg-neutral-900 p-20 text-center rounded-lg"
      >
        <div
          class="flex h-16 justify-center"
        >
          <img
            src="[object Object]"
          />
        </div>
        <p
          class="text-white"
        >
          Dashboard Login
        </p>
        <input
          class="h-10 outline-none bg-neutral-700 text-center text-white rounded-md"
          placeholder="Id"
        />
        <input
          class="h-10 outline-none bg-neutral-700 text-center text-white rounded-md"
          placeholder="password"
        />
        <button
          class="bg-indigo-900 h-10 rounded-md text-white text-sm text-center"
        >
          Login
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
